tests/auth_plugin_test.php

```diff
@@ -708,4 +709,33 @@ public function test_that_return_wantsurl_if_it_is_external_url() {
         $this->assertEquals('http://test.com/course/index.php?id=12&key=134', $redirect);
     }
 
+    /**
+     * Test that login page hook redirects correctly.
+     */
+    public function test_loginpage_hook_redirects_correctly() {
+        global $SESSION;
+
+        $SESSION->enrolkey_skipsso = 0;
+        set_config('ssourl', 'http://google.com', 'auth_userkey');
+        $this->auth = new auth_plugin_userkey();
+
+        $userredirect = $this->auth->should_login_redirect();
+        $this->assertEquals($userredirect, true);
+
+    }
+
+    /**
+     * Test that Moodle login page is displayed if url param is set correctly.
+     */
+    public function test_loginpage_hook_displays_login_correctly() {
+        global $SESSION;
+
+        $SESSION->enrolkey_skipsso = 1;
+        set_config('ssourl', 'http://google.com', 'auth_userkey');
+        $this->auth = new auth_plugin_userkey();
+        $userredirect = $this->auth->should_login_redirect();
+        $this->assertEquals($userredirect, false);
```

Comment:
As an option you can wrap redirection in a function and test that this function gets called in a hook method with correct data. We need tests for public hook function. Otherwise tests are sort of useless.